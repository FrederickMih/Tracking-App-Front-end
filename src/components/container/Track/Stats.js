import React from 'react';
import PropTypes from 'prop-types';
import '../../../styles/Stats.css';

const Stats = ({ today }) => (
  <div className="stats">
    <div className="stats-item">
      <div>Total</div>
      <div>
        {today}
        {' '}
        cm
      </div>
    </div>
  </div>
);
Stats.propTypes = {
  today: PropTypes.number.isRequired,
};

export default Stats;
