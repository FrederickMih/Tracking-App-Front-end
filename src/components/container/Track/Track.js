import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link, Redirect } from 'react-router-dom';
import NavigateNextIcon from '@material-ui/icons/NavigateNext';
import Header from '../Header';
import {
  getAllMeasures,
  getMeasuresStatus,
  loadMeasures,
} from '../../../Redux/slicers/measure';
import '../../../styles/Track.css';
import Profile from '../Profile';

const Track = () => {
  const title = 'Measure Tracked';
  const dispatch = useDispatch();
  const allMeasures = useSelector(getAllMeasures);
  const measures = useSelector(getMeasuresStatus);

  useEffect(async () => {
    await dispatch(loadMeasures());
  }, []);

  if (measures.status === 401) {
    return <Redirect to="/" />;
  }

  return (
    <div>
      <Header title={title} />
      <div className="track">
        <Profile />
        {Object.entries(allMeasures).length === 0 ? (
          <div className="not-found">
            No data found. Please Add Measure first
          </div>
        ) : (
          Object.entries(allMeasures).map((m) => (
            <div key={m[0]} className="track-tile">
              <Link to={{ pathname: '/track_details', info: m[1], date: m[0] }}>
                <div className="track-tile-left">
                  <div className="date">{m[0]}</div>
                  <div className="count">
                    Measured Parts:
                    {' '}
                    <span>{m[1].length}</span>
                  </div>
                </div>
                <div className="track-tile-right">
                  <span>Details</span>
                  <NavigateNextIcon />
                </div>
              </Link>
            </div>
          ))
        )}
      </div>
    </div>
  );
};

export default Track;
