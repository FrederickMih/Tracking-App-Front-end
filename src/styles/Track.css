.track {
  margin-top: 3em;
  height: calc(100vh - 7em);
  padding: 0.2em 0;
}

.not-found {
  font-size: 1.1rem;
  margin-top: 1em;
  text-align: center;
  color: var(--accent-primary);
  font-weight: bold;
  background-color: #fff;
  padding: 0.5em;
}

.track-tile {
  margin: 0.8em 0;
  padding: 0.4em 0.8em;
  background-color: #fff;
}

.track-tile a {
  text-decoration: none;
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: var(--accent-secondary);
  transition: all linear 300ms;
}

.track-tile a:hover {
  transform: scaleX(1.01);
}

.track-tile-left .date {
  font-size: 1.7rem;
  font-weight: bold;
  color: black;
}

.track-tile-left .count {
  color: rgb(33, 51, 70);
}

.track-tile-right {
  font-weight: bold;
  font-size: 0.9rem;
  display: flex;
  color: teal;
}

.track-tile-right > * {
  margin-left: 0.4em;
}

.track-tile-right span {
  margin-left: 0.4em;
  margin-top: 0.1em;
}
