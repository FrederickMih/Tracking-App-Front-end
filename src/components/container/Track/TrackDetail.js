import React, { useEffect } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import Header from '../Header';
import {
  getMeasurements,
  loadMeasurements,
} from '../../../Redux/slicers/measurement';
import '../../../styles/TrackDetails.css';
import Stats from './Stats';

const TrackDetails = ({ location }) => {
  const { info, date } = location;
  const title = 'Track Measure Details';
  const dispatch = useDispatch();
  const measurementsObj = useSelector(getMeasurements);
  const measurements = measurementsObj.measurements === undefined
    ? []
    : measurementsObj.measurements;
  const todayTotal = info ? info.map((e) => e.data).reduce((a, b) => a + b) : 0;

  useEffect(() => {
    dispatch(loadMeasurements());
  }, []);

  if (measurementsObj.status === 401) {
    return <Redirect to="/" />;
  }
  return (
    <>
      <Header title={title} link="/track" />
      <div className="details">
        <div className="details-header">
          <div className="date">{date}</div>
          <Stats today={todayTotal || 0} />
        </div>
        <div className="details-items">
          {info
            && info.map((e) => (
              <div key={e.id} className="details-item">
                <div className="ins-type">
                  {measurements.length > 0
                    ? measurementsObj.measurements.filter(
                      (f) => f.id === e.measurement_id
                    )[0].measurement_name
                    : ''}
                </div>
                <div className="data">
                  {e.data}
                  {' '}
                  <span>CM</span>
                </div>
              </div>
            ))}
        </div>
      </div>
    </>
  );
};

TrackDetails.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default TrackDetails;
