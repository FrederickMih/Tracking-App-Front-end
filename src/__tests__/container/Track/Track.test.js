import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../Redux/configureStore';
import Track from '../../../components/container/Track/Track';

const store = configureAppStore();

describe('Track', () => {
  it('should match the snapshot of Track component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Track />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
