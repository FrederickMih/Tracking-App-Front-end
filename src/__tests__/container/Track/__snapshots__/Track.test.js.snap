// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track should match the snapshot of Track component 1`] = `
<div>
  <div
    className="headline"
  >
    Measure Tracked
    <div
      className="bkbar"
    >
      <a
        href="/add"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div
    className="track"
  >
    <div
      className="profile-top"
    >
      <h2>
        Hi Â 
        <span
          className="profile-cap"
        >
           
        </span>
      </h2>
    </div>
    <button
      className="logout-button"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="logout"
        className="logoutBtn__color"
        src="logout.png"
      />
    </button>
    <div
      className="not-found"
    >
      No data found. Please Add Measure first
    </div>
  </div>
</div>
`;
